### PR TITLE
Add the scenario (“BLM”) chart

### PR DESCRIPTION
**Pull request by clementprdhomme**
### Overview

This PR adds a new component, `BlmChart`, that corresponds to the chart displayed when the user is creating a scenario.

<p align="center">
Line chart with thumbnails attached to some points
</p>

When the user hovers a point, only that point has its thumbnail shown. If some points don't have a thumbnail, hovering them will change nothing.

By default, the thumbnails of the first and last point, as well as the one of the “BLM” point are shown. Similarly to above, if one of these points doesn't have a thumbnail, it will not be shown by default.

### Designs

The chart can be found on [this screen](https://www.figma.com/file/hq0BZNB9fzyFSbEUgQIHdK/Marxan-Visual_V02?node-id=1523%3A3915) on Figma.

There are two elements of the design that I haven't implemented:
1. The blue area below the lines should have rounded corners. I'm unsure how to implement this.
2. The same with the lines that connect the points to their thumbnails: they should have rounded corners.

I'd like to point out that the algorithm that positions the thumbnails assumes the points draw a curve similar to `f: x => 1 / x`, when `x > 0`. If the curve is different, the thumbnails could be displayed on top of each other, on top of the line or even outside the container.

I'm also assuming the points will be fairly distributed along the curve. If they are not, the same could happen.

There's another edge case that could compromise the chart visually: if the BLM point is too high, the text “Recommended (BLM)” could be on top of “Boundary length”, as it is absolutely positioned.

### Testing instructions

Here's what you can test:
1. The chart visually complies with the design
2. If you add or remove the thumbnail of points, when hovering them, the behaviour should be the one described above
3. The chart should respond to changes in width/height with grace

### CODE REVIEW

**Review by mbarrenechea — APPROVED** [review]
GOOD JOB:
- About rounding paths, there is a very good post in [StackOverflow](https://stackoverflow.com/questions/10177985/svg-rounded-corner) about it. If you want to try, it's up to you. We can also talk with the designers and tell them that's not possible. IMO, it's good at it is.
- Regarding the text “Recommended (BLM)” could be on top of “Boundary length”, as it is absolutely positioned. We need to talk with the designers to find a solution that will always work. It's more a design problem than a development one

File: app/components/scenarios/blm-chart/component.tsx [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+            <text
+              x={Y_AXIS_WIDTH}
+              y="0"
+              textAnchor="end"
+              className="transform translate-y-3 text-xs text-black opacity-50"
+            >
+              More
+            </text>
+          </g>
+          <g transform={`translate(${0} ${yScale(yDomain[1]) + 0.3 * (yScale(yDomain[0]) - yScale(yDomain[1]))})`}>
```
[the remark this answers is not in the document]
Comment (clementprdhomme) [reply app/components/scenarios/blm-chart/component.tsx:163]: The “Boundary length” text is located at 30% of the top of the Y axis.

**Comment by clementprdhomme** [comment]
I've pushed two new changes to the PR:
- The blue area now has rounded corner at its bottom (thanks to clipping)
- The links between the points and the thumbnails are also rounded (thanks to a custom path generator)

<p align="center">
Same chart, but with rounded corners at the bottom of the blue area that is below the line, and rounded links that connect the points to their thumbnails
</p>

Regarding the overlapping of “Boundary length” and “Recommended (BLM)”, I don't think further changes are needed at this point. Alicia told me the curve should always look like it does, and that the BLM point should be located close to its inflexion. If this is not true, more than one assumption will be broken.

File: app/components/scenarios/blm-chart/component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbarrenechea) [reply app/components/scenarios/blm-chart/component.tsx:176]: ok, as you said this graph is going to be always the same

**Review by mbarrenechea — APPROVED** [review]
Good job

**Pull request merged**